### PR TITLE
agat: native keyboard, video and floppy improvements

### PR DESCRIPTION
**Pull request by shattered**

### CODE REVIEW

File: src/mame/machine/agatkeyb.h [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+***************************************************************************/
+
+DECLARE_DEVICE_TYPE(AGAT_KEYBOARD, agat_keyboard_device)
+
+
+/***************************************************************************
+    REUSABLE I/O PORTS
+***************************************************************************/
+
+INPUT_PORTS_EXTERN( agat_keyboard );
```
Comment (cuavas) [inline src/mame/machine/agatkeyb.h:22]: Is this key matrix actually likely to be used by things elsewhere, or would it be better to keep it confined to one compilation unit?

File: src/mame/machine/agatkeyb.h [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+		const machine_config &mconfig,
+		device_type type,
+		char const *tag,
+		device_t *owner,
+		u32 clock);
+	virtual void device_start() override;
+	virtual void device_reset() override;
+	virtual void key_make(u8 row, u8 column) override;
+	virtual void key_repeat(u8 row, u8 column) override;
+	virtual void send_key(u16 code);
```
Comment (cuavas) [inline src/mame/machine/agatkeyb.h:64]: Does this really need to be virtual?  Are there more keyboards in the family that will only differ in how they send the codes?

**Review by cuavas — COMMENTED** [review]
Looks good.  Just a couple of questions.

File: src/mame/machine/agatkeyb.h (hunk 2 above)
Comment (shattered) [reply src/mame/machine/agatkeyb.h:64]: Probably not.   There are at least two more keyboards (almost certainly with different matrix), but I'm not sure they're interesting enough to emulate.

File: src/mame/machine/agatkeyb.h (hunk 1 above)
Comment (shattered) [reply src/mame/machine/agatkeyb.h:22]: I don't think any other drivers would use this keyboard...

Comment (cuavas) [reply src/mame/machine/agatkeyb.h:22]: Better to not declare it `extern` – keep it contained and make it `static`.

File: src/mame/machine/agatkeyb.h (hunk 2 above)
Comment (cuavas) [reply src/mame/machine/agatkeyb.h:64]: Better not to make unnecessary vtable entries then.  The translate function probably doesn’t need to be virtual either.

**Pull request merged**